src/test/groovy/io/github/cosmicsilence/scalafix/ScalafixPluginTest.groovy

```diff
@@ -335,8 +335,10 @@ class ScalafixPluginTest extends Specification {
         ScalafixTask scalafixTestTask = tasks.scalafixTest // forces plugin configuration
         ScalaCompile compileMainTask = tasks.compileScala
         ScalaCompile compileTestTask = tasks.compileTestScala
-        [compileMainTask, compileTestTask].each { task -> // emulates the task execution
-            task.actions.find { it.displayName.contains('doFirst') }.execute(task)
+
+        [tasks.configureMainSemanticdbCompilerPlugin, tasks.configureTestSemanticdbCompilerPlugin].each { task -> // emulates the task execution
+            assert task.onlyIf.specs.get(1).isSatisfiedBy(task)
+            task.run()
```

Comment:
I'm thinking if we could simplify some of these tests by introducing a test class for the new task and moving everything concerning the Semanticdb setup into it. Tests in this class would just check if the dependencies between tasks were created.
Also, there are some tests in `ScalafixPluginFunctionalTest` that may need to be updated